csharp/extractor/Semmle.Extraction/Context.cs

```diff
@@ -200,7 +215,17 @@ public void AddFreshLabel(IEntity entity)
 #if DEBUG_LABELS
         readonly Dictionary<string, ICachedEntity> idLabelCache = new Dictionary<string, ICachedEntity>();
 #endif
-        readonly Dictionary<object, ICachedEntity> objectEntityCache = new Dictionary<object, ICachedEntity>();
+        class SymbolComparer : IEqualityComparer<object>
+        {
+            IEqualityComparer<ISymbol> comparer = SymbolEqualityComparer.IncludeNullability;
+
+            bool IEqualityComparer<object>.Equals(object x, object y) => comparer.Equals((ISymbol)x, (ISymbol)y);
+
+            int IEqualityComparer<object>.GetHashCode(object obj) => comparer.GetHashCode((ISymbol)obj);
+        }
+
+        readonly IDictionary<object, ICachedEntity> objectEntityCache = new Dictionary<object, ICachedEntity>();
+        readonly IDictionary<object, ICachedEntity> symbolEntityCache = new Dictionary<object, ICachedEntity>(10000, new SymbolComparer());
```

Comment:
We should be able to change `object` to `ISymbol` here and in `SymbolComparer`, right?

Reply:
I originally had some difficulty converting an `IDictionary<ISymbol, ICachedEntity>` to `IDictionary<object, ICachedEntity>` but managed to work around it in the end. The result is better, so thanks!

Reply:
🎉